bundles/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/handler/UniFiAccessPointThingHandler.java

```diff
@@ -0,0 +1,122 @@
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An access point managed by the UniFi controller software.
+ *
+ * @author Thomas Lauterbach - Initial contribution
+ */
+@NonNullByDefault
+public class UniFiAccessPointThingHandler extends UniFiBaseThingHandler<UniFiDevice, UniFiAccessPointThingConfig> {
+
+    private final Logger logger = LoggerFactory.getLogger(UniFiAccessPointThingHandler.class);
+
+    private UniFiAccessPointThingConfig config = new UniFiAccessPointThingConfig();
+
+    public UniFiAccessPointThingHandler(final Thing thing) {
+        super(thing);
+    }
+
+    private static boolean belongsToSite(final UniFiDevice client, final String siteName) {
+        boolean result = true;
+        if (!siteName.isEmpty()) {
+            final UniFiSite site = client.getSite();
+            if (site == null || !site.matchesName(siteName)) {
+                result = false;
+            }
+        }
+        return result;
+    }
+
+    @Override
+    protected boolean initialize(final UniFiAccessPointThingConfig config) {
+        this.config = config;
+        if (!config.isValid()) {
+            // ToDo Error message
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/error.thing.ap.offline.configuration_error");
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    protected @Nullable UniFiDevice getEntity(final UniFiControllerCache cache) {
+        final UniFiDevice device = cache.getDevice(config.getMacAddress());
+        if (device == null || !belongsToSite(device, config.getSite()) || !DEVICE_TYPE_UAP.equals(device.getType())) {
+            return null;
+        }
+        return device;
+    }
+
+    @Override
+    protected State getChannelState(final UniFiDevice device, final String channelId) {
+        State state = getDefaultState(channelId);
+
+        switch (channelId) {
+            case CHANNEL_AP_ENABLE:
+                state = OnOffType.from(!device.isDisabled());
+                break;
+        }
+        return state;
+    }
+
+    @Override
+    protected boolean handleCommand(final UniFiController controller, final UniFiDevice device,
+            final ChannelUID channelUID, final Command command) throws UniFiException {
+        final String channelID = channelUID.getIdWithoutGroup();
+
+        switch (channelID) {
+            case CHANNEL_AP_ENABLE:
+                if (command instanceof OnOffType) {
+                    return handleEnableCommand(controller, device, channelUID, command);
+                }
+                break;
+            default:
+                return false;
+        }
+        return false;
```

Comment:
Suggestion to simplify code:
```suggestion
        if (CHANNEL_AP_ENABLE.equals(channelID) && command instanceof OnOffType onOffCommand) {
            return handleEnableCommand(controller, device, channelUID, onOffCommand);
        }
        return false;
```
See other comment related to usage of `onOffCommand` variable.